### ISSUE TITLE
Problem with existsIn rule when using Translate behavior

### ISSUE THREAD

**Issue by segy** [cakephp/cakephp#5877]
I have a table in a plugin defined this way:

```php
class CategoriesTable extends Table
{
    public function initialize(array $config)
    {
        parent::initialize($config);

        // add Translate behavior
        $this->addBehavior('Translate', [
            'fields' => ['name'],
            'translationTable' => 'PluginI18n'
        ]);

        $this->belongsTo('Parent', [
            'className' => 'Plugin.Categories',
            'foreignKey' => 'parent_id'
        ]);

        $this->hasMany('Child', [
            'className' => 'Plugin.Categories',
            'foreignKey' => 'parent_id'
        ]);
    }

    /**
     * Default validation rules.
     *
     * @param  \Cake\Validation\Validator $validator Validator instance.
     * @return \Cake\Validation\Validator
     */
    public function validationDefault(Validator $validator)
    {
        $validator
            ->add('parent_id', 'valid', ['rule' => 'numeric'])
            ->allowEmpty('parent_id')
            ->requirePresence('name', 'create')
            ->notEmpty('name')
            ->add('active', 'valid', ['rule' => 'boolean']);

        return $validator;
    }

    /**
     * Returns a rules checker object that will be used for validating
     * application integrity.
     *
     * @param  \Cake\ORM\RulesChecker $rules The rules object to be modified.
     * @return \Cake\ORM\RulesChecker
     */
    public function buildRules(RulesChecker $rules)
    {
        $rules->add($rules->existsIn(['parent_id'], 'Parent'));

        return $rules;
    }
}
```

When I try to save an entity with not null parent_id, I get this error:
*Error: SQLSTATE[23000]: Integrity constraint violation: 1052 Column 'id' in where clause is ambiguous*

```sql
SELECT 1 AS `existing` FROM plugin_categories Parent LEFT JOIN plugin_i18n Parent_name_translation ON (Parent_name_translation.model = :c0 AND Parent_name_translation.field = :c1 AND Parent_name_translation.locale = :c2 AND Parent.id = (Parent_name_translation.foreign_key)) WHERE id = :c3 LIMIT 1
```

Testing controller code is:

```php
I18n::locale('sk_SK');
$cat = $this->Categories->newEntity();
$cat = $this->Categories->patchEntity($cat, ['name' => 'Test', 'parent_id' => 1, 'active' => 1]);
var_dump($this->Categories->save($cat));

**Comment by segy** [comment]
one more thing - not sure if you fixed it too. the same problem is for unique validation

```php
$validator->add('configuration_key', ['unique' => ['rule' => 'validateUnique', 'provider' => 'table']])
```

**Comment by markstory** [comment]
Thanks @segy I'll take a look.

**Comment by markstory** [comment]
Fixes for IsUnique added to #5944
